src/createEffectCapableStore.js

```diff
@@ -68,13 +13,67 @@ export const wrapReplaceReducer = store => nextReducer =>
  * @returns {Function} Store factory
  */
 export default createStore => (rootReducer, initialState) => {
-  const store = createStore(enhanceReducer(rootReducer), new AppStateWithEffects(initialState, []));
+  // Reference to the store dispatch function.
+  let dispatch;
+
+  const cleanTask = new Task();
+  const effectQueue = [];
+
+  /**
+   * Takes all of the effects in the closure effect queue, dispatches them,
+   * captures their values and empties the queue.
+   *
+   * @private
+   * @returns {Array<any>} The return values of dispatch.
+   */
+
+  const cleanEffectQueue = () => {
+    const values = effectQueue.map(dispatch);
+    effectQueue.length = 0; // @see https://davidwalsh.name/empty-array
+    return values;
+  };
+
+  /**
+   * Defers an effect to be dispatched by `cleanEffectQueue` later. Does this by
+   * adding the effect to an internal queue and deferring the `cleanEffectQueue`
+   * function if no other task is set.
+   *
+   * @private
+   * @param {any} The effect action to be dispatched later.
+   */
+
+  const deferEffect = effect => {
+    effectQueue.push(effect);
```

Comment:
I'm not sure why we would want to move this out of the closure. It seems like a lot more code in currying then is necessary, plus [redux](https://github.com/rackt/redux/blob/master/src/createStore.js) uses this pattern.